cmake/Modules/FindComputeCpp.cmake

```diff
@@ -259,19 +262,25 @@ function(__build_ir)
     set(device_compiler_cxx_standard "")
   endif()
 
+  list(APPEND COMPUTECPP_DEVICE_COMPILER_FLAGS
+    ${device_compiler_cxx_standard}
+    ${COMPUTECPP_USER_FLAGS}
+  )
+
   get_property(source_compile_flags
     SOURCE ${SDK_BUILD_IR_SOURCE}
     PROPERTY COMPUTECPP_SOURCE_FLAGS
   )
+  separate_arguments(source_compile_flags)
   if(source_compile_flags)
-    list(APPEND target_compile_flags ${source_compile_flags})
+    list(APPEND COMPUTECPP_DEVICE_COMPILER_FLAGS ${source_compile_flags})
   endif()
 
-  list(APPEND COMPUTECPP_DEVICE_COMPILER_FLAGS
-    ${device_compiler_cxx_standard}
-    ${COMPUTECPP_USER_FLAGS}
-    ${target_compile_flags}
-  )
+  get_target_property(target_compile_flags ${SDK_BUILD_IR_TARGET} COMPILE_FLAGS)
```

Comment:
Would it be better to use a generator expression to extract the compile flags? That way a user can add flags to the target after calling `add_sycl_to_target` and they will still be added to the custom command.

Reply:
That's a good point, it would be good yeah. I'm not too comfortable with generators but I can try that.

Reply:
I tried something similar to the 2 generators above but I can't make it work if the property is a space separated string (which it is in Eigen). It is considered as a single argument so the `join` does nothing and once it is given to `add_custom_command` CMake will backslash the spaces...
There is a similar issue here: http://www.gccxml.org/Bug/view.php?id=14353#c33712
The only solution I see is to write all the arguments to a file and tell the compiler to read the arguments from there.
Note that the 2 others generators would not work with a space separated property either they just don't happen to be used in Eigen. Is Eigen in the wrong here or is this something we should support?

Reply:
The option should be a semicolon separated cmake list, not space separated. That sounds like an Eigen problem to me.

Also I just noticed this is using `COMPILE_FLAGS` --- this is deprecated and you should be using `COMPILE_OPTIONS`. I have no idea if that works if Eigen is setting `COMPILE_FLAGS` though.

Reply:
Ok I could convert the COMPILE_FLAGS string to a COMPILE_OPTIONS list just in Eigen only for SYCL so that's working fine, thanks.